openslides_backend/action/action_worker.py

```diff
@@ -0,0 +1,288 @@
+                    "action_data_error_index": 0,
+                }
+            self.datastore.write_action_worker(
+                WriteRequest(
+                    events=[
+                        Event(
+                            type=EventType.Update,
+                            fqid=self.fqid,
+                            fields={
+                                "state": state,
+                                "timestamp": current_time,
+                                "result": response,
+                            },
+                        )
+                    ],
+                    information={self.fqid: ["finish action_worker"]},
+                    user_id=self.user_id,
+                    locked_fields={},
+                )
+            )
+            self.logger.debug(
+                f"action_worker '{self.fqid} {self.action_names}': {current_time}"
+            )
+
+
+class ActionWorker(threading.Thread):
+    def __init__(
+        self,
+        payload: Payload,
+        user_id: int,
+        is_atomic: bool,
+        handler: ActionHandler,
+        lock: threading.Lock,
+        thread_executor_pid: int,
+        action_worker_writing: ActionWorkerWriting,
+    ) -> None:
+        super().__init__(name="action_worker")
+        self.handler = handler
+        self.payload = payload
+        self.user_id = user_id
+        self.is_atomic = is_atomic
+        self.lock = lock
+        self.thread_executor_pid = thread_executor_pid
+        self.action_worker_writing = action_worker_writing
+        self.started: bool = False
+
+    def run(self):  # type: ignore
+        with self.lock:
+            self.started = True
+            try:
+                self.response = self.handler.handle_request(
+                    self.payload, self.user_id, self.is_atomic
+                )
+            except Exception as exception:
+                self.exception = exception
+
+
+def gunicorn_post_request(
+    worker: ThreadWorker, req: Request, environ: Dict[str, Any], resp: Response
+) -> None:
+    """
+    gunicorn server hook, called after response of one request
+    was send to client in the Thread of Gunicorns ThreadPool.
+
+    if the resp.status_code is 202 (HTTPStatus.ACCEPTED.value) there is an
+    action_thread created, which wasn't finished before the response
+    to the client was send and should be kept alive until it ends.
+
+    The corresponding action_worker-thread will be found by it's
+    thread executor_pid.
+    """
+    if resp.status_code != HTTPStatus.ACCEPTED.value:
+        return
+    action_worker: Optional[ActionWorker] = None
+    curr_thread = threading.current_thread()
+    for thread in threading.enumerate():
+        if (
+            thread.name == "action_worker"
+            and cast(ActionWorker, thread).thread_executor_pid == curr_thread.native_id
+        ):
+            action_worker = cast(ActionWorker, thread)
+            break
+
+    if not action_worker:
```

Comment:
```suggestion
    else:
```
finally a meaningful usage! :D